### PR TITLE
nixos/modules/services/web-apps/miniflux: fix permissions for PostgreSQL 15

### PR DESCRIPTION
**Pull request by ajaxbits**
In 23.11 and above, `services.postgresql` is going to [default to Postgres 15](https://github.com/NixOS/nixpkgs/commit/e0379772393d14090d1b112a7340bfefe580b0c9).  Postgres 15 [removes `CREATE` permissions](https://www.postgresql.org/about/news/postgresql-15-released-2526/) for all users on the `public` schema. 

Miniflux [needs `CREATE` permissions](https://github.com/miniflux/v2/blob/1bd5d57884a9a812ae8e703ca5a5502a702b9551/internal/database/migrations.go#L16) in order to do migrations. This PR adds that capability back.

Associated NixOS tests now pass successfully.

Part of ZHF: #265948

## Description of changes

- Makes the miniflux `${dbUser}` the owner of the `${dbName}` database. Giving them all necessary permissions.
- Adds a maintainer to the associated NixOS test.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`) [Made sure it worked with `nixos-shell`].
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ambroisie** [comment]
This is not the safest way to fix the issue. I think `gitea` has a better fix already merged (can't link it as I'm on mobile).

File: nixos/modules/services/web-apps/miniflux.nix [hunk 1]
```diff
@@ -13,6 +13,7 @@ let
   preStart = pkgs.writeScript "miniflux-pre-start" ''
     #!${pkgs.runtimeShell}
     ${pgbin}/psql "${dbName}" -c "CREATE EXTENSION IF NOT EXISTS hstore"
+    ${pgbin}/psql "${dbName}" -c "ALTER DEFAULT PRIVILEGES IN SCHEMA public GRANT ALL PRIVILEGES ON TABLES TO ${dbUser};"
```
Comment (ambroisie) [inline nixos/modules/services/web-apps/miniflux.nix:16]:
```suggestion
    ${pgbin}/psql "${dbName}" -c "ALTER DATABASE ${dbName} OWNER TO ${dbUser};"
```

Taken from #262741 (it was Forgejo, not Gitea)

File: nixos/modules/services/web-apps/miniflux.nix [hunk 2]
```diff
@@ -73,6 +74,8 @@ in
         name = dbUser;
         ensurePermissions = {
           "DATABASE ${dbName}" = "ALL PRIVILEGES";
+          "SCHEMA public" = "CREATE";
+          "ALL TABLES IN SCHEMA public" = "ALL PRIVILEGES";
```
Comment (ambroisie) [inline nixos/modules/services/web-apps/miniflux.nix:78]: And this shouldn't be needed with the `preStart` script.

**Review by ambroisie — CHANGES_REQUESTED** [review]

**Comment by ajaxbits** [comment]
> This is not the safest way to fix the issue. I think `gitea` has a better fix already merged (can't link it as I'm on mobile).

@ambroisie Thank you so much for the feedback! I'll look into all those resources. 

This is my first major PR to `nixpkgs`, and maybe I opened the PR too soon, since this was still very much WIP. So I'm sorry if I wasted any of your time -- I'll implement those suggestions and confirm all the tests pass before marking as ready for review.

File: nixos/modules/services/web-apps/miniflux.nix (hunk 2 above)
Comment (ajaxbits) [reply nixos/modules/services/web-apps/miniflux.nix]: Removed!

**Comment by ambroisie** [comment]
No problem, reviews and feedback is how we all learn.

I was just looking to fix the same issue on a different package, which is why I was on the lookout.

**Comment by ajaxbits** [comment]
Sick! Well thanks for being on the lookout!

Looks like tests are running nicely now. I'll be able to clean up commit messages and PR descriptions here in a bit and get this up for review soon.

**Comment by ajaxbits** [comment]
Result of `nixpkgs-review pr 266511` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

**Review by ambroisie — APPROVED** [review]

**Comment by ajaxbits** [comment]
IMO, we should go with #266270 if it gets merged. But if that's not merged before the 23.11 cutoff, we should merge this.

**Comment by Artturin** [comment]
https://github.com/NixOS/nixpkgs/pull/266270 was merged

**Pull request closed without merging**